extract/extract/parse.py

```diff
@@ -66,3 +71,22 @@ def extract_schema_org(page_html: str) -> dict:
     except JSONDecodeError:
         schema_org = extruct.extract(page_html, syntaxes=_SYNTAXES)
     return schema_org if schema_org else {}
+
+
+def extract_schema_org_hm(page_html: str) -> dict:
+    """
+    Extract schema.org information form `page_html`.
+
+    Args:
+        page_html (str): HTML of the page
+
+    Returns:
+        dict: Schema.org information found in `page_html`
+    """
+    unescaped_html = html.unescape(page_html)
+    try:
+        schema_org = extruct.extract(unescaped_html, syntaxes=_SYNTAXES)
+    except JSONDecodeError:
+        unescaped_html = decode(page_html.encode("utf-8"), "unicode-escape")
+        schema_org = extruct.extract(unescaped_html, syntaxes=_SYNTAXES, errors="ignore")
```

Comment:
No, for otto it doesn't help. But instead of a seperate method, I added now another try-except block which is using h&m specific encoding-decoding if none of the other approaches work. https://github.com/calgo-lab/green-db/blob/f5d34655d29ef5de9120d2d0de3000eeb2c3d83b/extract/extract/parse.py#L68-L76

Reply:
It should be possible to use `page_html.encode("utf-8").decode("unicode-escape")` instead of importing `decode`.

Reply:
I just checked it and the tests were running fine 👍